Create a seldon-serve ksonnet prototype

ankushagarwal opened the pull request:
This prototype can be used to create components which contain easily
deployable "seldondeployment" objects. The use case for this is that
once a user wraps a seldon-core model into a docker image, the user can
create a "seldondeployment" using this component.

It contains the following params: namespace, name, image, replicas.

We can add more params later.

An example for using this would be

Create a component:
ks generate seldon-serve issue-summarization --name=issue-summarizatiion
--image=gcr.io/kubeflow-images-staging/issue-summarization-seldon:20180312
--replicas=3

Apply the component
ks apply env -c issue-summarization.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/kubeflow/413)
<!-- Reviewable:end -->

ankushagarwal commented:
/cc @cliveseldon

cliveseldon commented:
@ankushagarwal  Looks interesting. The main question is what is easiest for user to have to build different graph types. For a single model then what you have done is good but what about more complex graphs? I suppose we simply add further prototypes for A/B tests, ensembles, multi-armed bandits etc., to build up a whole library of different prototypes for normal tasks. 

Also:
  * The user might want to specify REST or GRPC for their model
  * it would be expected the user would then edit the PodTemplateSpec in "componentSpec" to add Volumes and other customizations needed.

Looks like a great start.

ankushagarwal commented:
Added support for specification of endpoint: REST or GRPC

Yes for things which cannot be easily parametrized such as volumes,  the user would be expected to edit the componentSpec directly.

Renamed the files to serve-simple.jsonnet and renamed the prototype to seldon-serve-simple

ankushagarwal commented:
My idea with this prototype was to be able to come up with an easy way to deploy a single model built using `seldonio/core-python-wrapper` without having to write the entire SeldonDeployment spec myself. 

I haven't given much thought to complex graph types. I guess it makes sense to have separate prototypes for the most common scenarios like: seldon-serve-ab-test, seldon-serve-ensemble, etc.

cliveseldon commented:
Looks good.

ankushagarwal commented:
Can you /lgtm so that I can go ahead and merge?

cliveseldon commented:
/lgtm

ankushagarwal commented:
/approve
